File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/solution.c
int numSteps(char* s) {
    int c=0,sum=0;
    int n=strlen(s);
    for(int i=n-1;i>0;i--)
    {
        int d=s[i]-'0';
        if(d+c==1)
        {
            c=1;
            sum+=2;
        }
        else{
            sum+=1;
        }

    }
    if(c==1)
    {
        sum+=1;
    }
    return sum;
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/solution.java
class Solution {
    public int numSteps(String s) 
    {
        int c=0,sum=0;
        for(int i=s.length()-1;i>0;i--)
        {
            int d=s.charAt(i)-'0';
            if(d+c==1)
            {
                c=1;
                sum+=2;
            }
            else{
                sum+=1;
            }
        }
        if(c==1)
        {
            sum+=1;
        }
        return sum;
    }
}
